### PR TITLE
executor: support limit sort

### PR DESCRIPTION
**Pull request by hanfei19910905**
This change can reduce the mem cost.

### CODE REVIEW

File: executor/executor_test.go [hunk 1]
```diff
@@ -29,6 +25,9 @@ import (
 	"github.com/pingcap/tidb/util/testkit"
 	"github.com/pingcap/tidb/util/testleak"
 	"github.com/pingcap/tidb/util/types"
+	"strings"
+	"testing"
+	"time"
```
Comment (coocood) [inline executor/executor_test.go]: We put go standard library packages above other packages, with an empty line to separate them.

File: executor/executor.go [hunk 2]
```diff
@@ -902,9 +903,18 @@ func (e *SortExec) Less(i, j int) bool {
 	return false
 }
 
+// SortBufferSize represents the total extra row count that sort can use.
+var SortBufferSize = 500
```
Comment (qiuyesuifeng) [inline executor/executor.go]: Seems unexported variable is OK.

Comment (coocood) [inline executor/executor.go]: This variable is exported for test, so we don't need to insert too many rows.

File: executor/executor_test.go (hunk 1 above)
Comment (hanfei19910905) [inline executor/executor_test.go]: i will fix it. btw, this is changed by fmt...

File: executor/executor.go (hunk 2 above)
Comment (hanfei19910905) [inline executor/executor.go]: and i think this value can be set by user in future :)

**Pull request merged**